### DISCUSSION TITLE
PC crashes on installation

### DISCUSSION THREAD

**Discussion by cratom4** [vladmandic/sdnext#999] [Q&A]
I was trying to re-install the webui (I was able to install the first time and used it for a while, though something messed up and decided to re-install it), when it comes to "installing torch torchvision", my computer immediately freezes and then crashes at the end.
I've searched and found adding "--no-cache-dir" could help, but not sure where in the code I should add.
I'm not even sure if this is right way of solving the problem, any help would be appreciated.

**Comment by vladmandic** [comment]
you didn't even say which system is it? mac, windows, linux?
in any case, check system logs for hints on what caused the crash, without looking into those, its impossible to say.
note that app runs in user-mode - there is no root or kernel access, so there is nothing that would justify system crash even in worst case scenarios - anything that causes crash would have to be related to device drivers that just happen to be used by app or something like that.

**Reply by cratom4** [reply]
System - windows10, 1070ti, 8gb vram, 8gb ram.
I'm not familiar with system logs, is this what you ask for.


- System 

  - Provider 

   [ Name]  Microsoft-Windows-Kernel-Power 
   [ Guid]  {331c3b3a-2005-44c2-ac5e-77220c37d6b4} 
 
   EventID 41 
 
   Version 8 
 
   Level 1 
 
   Task 63 
 
   Opcode 0 
 
   Keywords 0x8000400000000002 
 
  - TimeCreated 

   [ SystemTime]  2023-05-16T10:49:38.2844642Z 
 
   EventRecordID 39529 
 
   Correlation 
 
  - Execution 

   [ ProcessID]  4 
   [ ThreadID]  8 
 
   Channel System 
 
   Computer AAA-PC 
 
  - Security 

   [ UserID]  S-1-5-18 
 

- EventData 

  BugcheckCode 159 
  BugcheckParameter1 0x3 
  BugcheckParameter2 0x0 
  BugcheckParameter3 0x0 
  BugcheckParameter4 0x0 
  SleepInProgress 0 
  PowerButtonTimestamp 0 
  BootAppStatus 0 
  Checkpoint 0 
  ConnectedStandbyInProgress false 
  SystemSleepTransitionsToOn 0 
  CsEntryScenarioInstanceId 0 
  BugcheckInfoFromEFI true 
  CheckpointStatus 0 
  CsEntryScenarioInstanceIdV2 0 
  LongPowerButtonPressDetected false

**Comment by vladmandic** [comment]
yes, that is one of them. on windows, its all in event logs. but you need to find the important one, i cannot go over all event logs for you.

**Reply by cratom4** [reply]
The one I pasted was flagged as important, icon with red circle with white cross in it.
Or do I have to look for something else?

**Reply by vladmandic** [reply]
yes, but it doesn't have any sufficient information, so look around it.

**Comment by Tillerz** [comment]
`[ Name] Microsoft-Windows-Kernel-Power` isn't that just a hint it crashed / unclear restart?

My guess is that it's doing some probing on hardware or something and it crashes on something like CUDA because of the drivers and/or the CUDA installation?

Do you have the latest display drivers installed and the CUDA package?

- https://www.nvidia.com/en-us/geforce/drivers/
- https://developer.nvidia.com/cuda-downloads/

**Reply by vladmandic** [reply]
yes, i agree - that's why @cratom4 should check other events around that one.
and yes, updating display driver is first thing i'd try as well.

**Comment by Baskhara** [comment]
The 8GB RAM could be the problem here. I have observed on a system with 8GB RAM when installing torch, the python process maxed out the physical RAM causing every other process to Swap/Paged out from RAM to Disk which also causes the computer to freeze. I even seen it swapped the graphics driver causing the screen to blink, but if you keep the computer running, it'll eventually finish.

One thing to note, this is just my assumption after observing the process from Task Manager in Windows which show 7.8 / 7.9 GB used memory before it completely freeze and how it jump from 3 GB to 7.8 GB the moment torch is installing.

**Comment by cratom4** [comment]
Hello, I couldn't figure out which log to look at and check for the problem, so I decided to use the method of adding "--no-cache-dir" to installer.py.
Installation completed without any problem.

Display driver was up to date, but for the CUDA toolkit, not sure.
When I run webui, log states toolkit detected, but I couldn't use "nvcc -V" to check the version.
Now I installed it from the link just to make sure. (I only checked post by Tillerz after the installation so I can't tell if this could have solved the problem)

And yes, I've seen same RAM usage as Baskhara mentioned, though in my case waiting didn't help sadly.

Anyways thank you all for your help.
